### ISSUE TITLE
Notify timestamps with no deadline nor scheduled

### ISSUE THREAD

**Issue by wlmb** [p-m/org-notify#11]
Is it possible to modify the code to allow notifications for items that have no deadline or are not scheduled but have an ordinary timestamp? I saw the variable my-org-notify-timestamp-type defines currently only the two types :deadline and :scheduled

**Comment by p-m** [comment]
On Wed, Sep 11 2024, W. Luis Mochán wrote:

> Is it possible to modify the code to allow notifications for items that have
> no deadline or are not scheduled but have an ordinary timestamp?

Hi,

Certainly yes. But first, you need to provide an example please.

Do you want something like this:

--8<---------------cut here---------------start------------->8---
* TODO An item with an ordinary timestamp
<2024-09-16 Mon 10:10>
--8<---------------cut here---------------end--------------->8---

?

-- 
           Peter

**Comment by wlmb** [comment]
Yes, exactly, an item with a timestamp but with no SCHEDULE or DEADLINE as in your example.

**Comment by p-m** [comment]
On Wed, Sep 11 2024, W. Luis Mochán wrote:

> Yes, exactly,

Ok. Then it seems to be difficult.

This is how the org-file is parsed:

--8<---------------cut here---------------start------------->8---
(with-current-buffer (find-file-noselect "filename.org")
  (org-element-parse-buffer 'headline))
--8<---------------cut here---------------end--------------->8---

The timestamp is not included in the result. If you find a way, to add
support for a naked timestamp, I could merge it perhaps.

Why do you need that?

-- 
           Peter

**Comment by wlmb** [comment]
Hi. I'll look in more detail at the code then.

According to the orgmode manual (for example, in https://orgmode.org/manual/Deadlines-and-Scheduling.html) ordinary timestamps are for appointments, meetings, etc., activities that are for a specific time and date, while SCHEDULE is for initiating work that may take a long time and DEADLINE for when it should be finished. Thus, my agenda is full of timestamps for meetings, classes, seminars, appointments, etc., with no DEADLINE and no SCHEDULE. It would be nice to be 'notified' of those events too. 

Timestamps are allowed in the body of an entry. Maybe that is the reason they are not parsed as part of the headlines.

Regards,
Luis

**Comment by wlmb** [comment]
Replacing
```
(org-element-parse-buffer 'headline)
```
with
```
(org-element-parse-buffer 'object)
```
does retrieve all the timelines as an element 
```
(paragraph ...(timestamp (:type active ...)))
```
after the headline. It also returns the scheduled and deadline entries within the headline as
``` 
(headline(... :deadline (timestamp(...)) :scheduled(timestamp(...))) 
  (section...(planning(... :deadline (timestamp(...))  :scheduled(timestamp(...)))))
```
Would this help?

Regards, 
Luis

**Comment by p-m** [comment]
On Thu, Sep 12 2024, W. Luis Mochán wrote:

> Replacing
>
> (org-element-parse-buffer 'headline)
>
> with
>
> (org-element-parse-buffer 'object)

Hi,
With this modification, org-notify-process takes about 6 times more
time on my system. So IMHO this should be customisable.


> does retrieve all the timelines as an element
>
> (paragraph ...(timestamp (:type active ...)))

Ok.


> Would this help?

Yes. But I don’t know, if and when I could do that.
If somebody provides a patch, it will be welcome.

-- 
           Peter

**Comment by wlmb** [comment]
I understand. I'll try to find some time to try to understand the required changes, but I don't think it will be too soon :(

**Comment by lnguyen4** [comment]
i don't know if this will help but org-mode has repeated-tasks

https://orgmode.org/manual/Repeated-tasks.html

This may be relevant to the events that are scheduled periodically. 

I am just about to try org-notify for my workflow so I am not really versed in the details of this package.
